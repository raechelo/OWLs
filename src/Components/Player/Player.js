import React, { Component } from 'react';
import sCrest from '../../assets/sly-crest.png';
import gCrest from '../../assets/gryf-crest.png';
import rCrest from '../../assets/rav.png';
import hCrest from '../../assets/hufflepuff.png';

export default class Player extends Component {
  constructor() {
    super()
    this.state = {
      hufflePoints: 0,
      slyPoints: 0,
      gryfPoints: 0,
      ravenPoints: 0
    }
  }

  render() {
    return (
      <section className="player-section">
        <article className="left-side">
          <h6 className="left-house">Gryffindor</h6>
          <h6 className="left-points">{this.state.gryfPoints} Points</h6>
        <img className="house-crest" src={gCrest} alt="Gryffindor House Crest" />
        </article>
        <article className="right-side">
          <img className="house-crest" src={sCrest} alt="Slytherin House Crest" />
          <h6 className="right-house">Slytherin</h6>
          <h6 className="right-points">{this.state.hufflePoints} Points</h6>
        </article>
        <article className="left-side bot">
          <h6 className="leftt-house">Ravenclaw</h6>
          <h6 className="left-points">{this.state.ravenPoints} Points</h6>
          <img className="house-crest" src={rCrest} alt="Ravenclaw House Crest" />
        </article>
        <article className="right-side bot">
          <img className="house-crest" src={hCrest} alt="Hufflepuff House Crest" />
          <h6 className="right-points">{this.state.slyPoints} Points</h6>
          <h6 className="left-house">Hufflepuff</h6>
        </article>
      </section>
    )
  }
}