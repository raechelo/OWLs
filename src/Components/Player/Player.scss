.house-crest {
  width: 100%;
}

.player-section {
  background-color: $boardBg;
  display: grid;
  grid-template-columns: 2fr 2fr;
  margin: 5rem auto;
  height: 25%;
  width: $boardW;

  h6 {
    display: inline;
    font-family: $bodyfont;
    font-size: 1.5rem;
    margin: auto auto;
  }


}

.left-side {
  display: grid;
  grid-template-areas: 
  "l-house l-crest"
  "l-points l-crest";
  grid-auto-columns: 2fr 1fr;

  .left-house {
    grid-area: l-house;
  }

  .left-points {
    grid-area: l-points;
  }

  img {
    grid-area: l-crest; 
  }


}
.right-side {
  display: grid;
  grid-template-areas: 
  "r-crest r-house"
  "r-crest r-points";
  grid-auto-columns: 1fr 2fr;

  .right-house {
    grid-area: r-house;
  }

  .right-points {
    grid-area: r-points;
  }

  img {
    grid-area: r-crest; 
  }
}