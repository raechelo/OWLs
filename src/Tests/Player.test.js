import React from 'react';
import Player from '../Components/Player/Player';
import { shallow } from 'enzyme';

describe('Player', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow (
      <Player />
    )
  })

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a proper default state', () => {
    expect(wrapper.state()).toEqual( {
      hufflePoints: 0,
      slyPoints: 0,
      gryfPoints: 0,
      ravenPoints: 0
    } )
  })
})