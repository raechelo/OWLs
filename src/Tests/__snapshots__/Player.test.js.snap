// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player should match the snapshot with all the data passed in 1`] = `
<section
  className="player-section"
>
  <article
    className="left-side"
  >
    <h6
      className="left-house"
    >
      Gryffindor
    </h6>
    <h6
      className="left-points"
    >
      0
       Points
    </h6>
    <img
      alt="Gryffindor House Crest"
      className="house-crest"
      src="gryf-crest.png"
    />
  </article>
  <article
    className="right-side"
  >
    <img
      alt="Slytherin House Crest"
      className="house-crest"
      src="sly-crest.png"
    />
    <h6
      className="right-house"
    >
      Slytherin
    </h6>
    <h6
      className="right-points"
    >
      0
       Points
    </h6>
  </article>
  <article
    className="left-side bot"
  >
    <h6
      className="leftt-house"
    >
      Ravenclaw
    </h6>
    <h6
      className="left-points"
    >
      0
       Points
    </h6>
    <img
      alt="Ravenclaw House Crest"
      className="house-crest"
      src="rav.png"
    />
  </article>
  <article
    className="right-side bot"
  >
    <img
      alt="Hufflepuff House Crest"
      className="house-crest"
      src="hufflepuff.png"
    />
    <h6
      className="right-points"
    >
      0
       Points
    </h6>
    <h6
      className="left-house"
    >
      Hufflepuff
    </h6>
  </article>
</section>
`;
